### DISCUSSION TITLE
Unable to read data from bluetooth device. A terminal app available on playstore is successfully receiving the data

### DISCUSSION THREAD

**Discussion by nickanimate** [kenjdavidson/react-native-bluetooth-classic#199] [Q&A] [answered]
I am using ESP32 as a bluetooth device. Running app on my Samsung phone with android 12.0. Functional components of React native are used. Device is successfully connecting and writing data but read is not working at all
I have tried separately - 
1. let a = await RNBluetoothClassic.readFromDevice(address)
2. let a = await connectedDevice.read()
3. RNBluetoothClassic.addEventListener(BluetoothEventType, readFn())
    readFn() => {
       let a = RNBluetoothClassic.readFromDevice(address);
       console.log(value);
    }

In all these cases I am receiving null values

**Comment by kenjdavidson** [comment]
Check some old issues regarding esp32. It's normally a delimiter issue. I'm on vaca and can't help until next week, but it should work with the correct delimiter.

/n is default
/n/r may be required

Get Outlook for Android<https://aka.ms/ghei36>
________________________________
From: nickanimate ***@***.***>
Sent: Monday, August 29, 2022 4:11:22 AM
To: kenjdavidson/react-native-bluetooth-classic ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [kenjdavidson/react-native-bluetooth-classic] Unable to read data from bluetooth device. A terminal app available on playstore is successfully receiving the data (Discussion #199)


I am using ESP32 as a bluetooth device. Running app on my Samsung phone with android 12.0. Functional components of React native are used. Device is successfully connecting and writing data but read is not working at all
I have tried separately -

  1.  let a = await RNBluetoothClassic.readFromDevice(address)
  2.  let a = await connectedDevice.read()
  3.  RNBluetoothClassic.addEventListener(BluetoothEventType, readFn())
readFn() => {
let a = RNBluetoothClassic.readFromDevice(address);
console.log(value);
}

In all these cases I am receiving null values

**Reply by nickanimate** [reply]
Hey! Adding Delimiter solved the issue. Thanks for the save!! Also, can we do something from front-end to read the data without delimiter?

**Comment by kenjdavidson** [comment, marked as answer]
Um. I think if you send a null or empty delimiter it will just send all the data to the end of what it has. I'm not 100 percent sure if I implemented that, but that was the plan.

Although it goes against my judgment of the name StringDelimitedDeviceCinnectionImpl.

Get Outlook for Android<https://aka.ms/ghei36>
________________________________
